.github/workflows/build.yml

```diff
@@ -4,16 +4,99 @@ on: [push, pull_request]
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest"]
+    runs-on: ${{ matrix.os }}
     steps:
     - uses: actions/checkout@v3
-
     - uses: ruby/setup-ruby@v1
       with:
-        ruby-version: 2.4.10
-
+        ruby-version: 3.2.2
     - name: Set up environment
       run: bundle install
-
     - name: Build
       run: bundle exec rake package
+      shell: bash
+    - name: Show standalone packages
+      run: ls pkg
+      shell: bash
+    - name: Upload standalone packages
+      uses: actions/upload-artifact@v3
+      with:
+        name: pkg
+        path: pkg
+  test:
+    needs: [build]
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["windows-latest","ubuntu-latest","macos-latest"]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Download all workflow run artifacts
+      uses: actions/download-artifact@v3
+    - name: test package
+      if: ${{ runner.os == 'MacOS'}}
+      run: |
```

Comment:
My preference is to keep as much code as possible in files that can be run locally, rather than putting lots of bash script lines in a workflow YAML file, because otherwise you end in the situation where you have to do a round trip push/build to test things, rather than getting local feedback.

Reply:
good shout, I had done it somewhere else locally on my machine but it was easier just to write one that find it again xD

have scripted up a hacky x-plat one